packages/salesforcedx-vscode-core/src/commands/functions/functionService.ts

```diff
@@ -180,27 +187,48 @@ export class FunctionService {
    */
   public async debugFunction(rootDir: string) {
     const functionExecution = this.getStartedFunction(rootDir);
-    if (functionExecution) {
-      const { debugPort, debugType } = functionExecution;
-      const debugConfiguration: vscode.DebugConfiguration = {
-        type: debugType,
-        request: 'attach',
-        name: 'Debug Invoke', // This name doesn't surface in UI
-        resolveSourceMapLocations: ['**', '!**/node_modules/**'],
-        console: 'integratedTerminal',
-        internalConsoleOptions: 'openOnSessionStart',
-        localRoot: rootDir,
-        remoteRoot: '/workspace',
-        hostName: '127.0.0.1',
-        port: debugPort
-      };
-      if (!functionExecution.debugSession) {
-        await vscode.debug.startDebugging(
-          getRootWorkspace(),
-          debugConfiguration
-        );
-      }
+    if (!functionExecution) {
+      // TODO: report error?
+      return;
+    }
+
+    if (!functionExecution.debugSession) {
+      const debugConfiguration = this.getDebugConfiguration(functionExecution, rootDir);
+
+      await vscode.debug.startDebugging(
+        getRootWorkspace(),
+        debugConfiguration
+      );
+    }
+  }
+
+  /***
+   * Create a DebugConfiguration object
+   */
+  public getDebugConfiguration(functionExecution: FunctionExecution, rootDir: string): vscode.DebugConfiguration {
+    const { debugPort, debugType } = functionExecution;
+    const debugConfiguration: vscode.DebugConfiguration = {
+      type: debugType,
+      request: 'attach',
+      name: 'Debug Invoke', // This name doesn't surface in UI
+      resolveSourceMapLocations: ['**', '!**/node_modules/**'],
+      console: 'integratedTerminal',
+      internalConsoleOptions: 'openOnSessionStart',
+      localRoot: rootDir,
+      remoteRoot: '/workspace',
+      hostName: '127.0.0.1',
+      port: debugPort
+    };
+
+    // The following check is specifically for when debugging a JavaScript function and not using
+    // a container (see W-10738230).  If there are issues with debugging TypeScript in the future,
+    // this is something to look into for TypeScript as well (though, at this time, debugging
+    // TypeScript w/o a container works).
+    if (functionExecution.isContainerLess && this.getFunctionType() === functionType.JAVASCRIPT) {
+      debugConfiguration.remoteRoot = undefined;
```

Comment:
Nit (as you already verified it works) but here would have been cleaner to delete the property instead of setting it to undefined. `delete debugConfiguration.remoteRoot`.  The diff being with the current flow we get 
```
{
      ...
      localRoot: rootDir,
      remoteRoot: undefined,
      ...
} 
```
where as if we use delete the property doesn't exist on the object at all.

Reply:
done